x/staking/client/testutil/helpers.go

```diff
@@ -23,74 +23,200 @@ func TxStakingCreateValidator(f *cli.Fixtures, from, consPubKey string, amount s
+	require.Empty(f.T, errStr)
 
-	var params types.Params
-
-	err := f.Cdc.UnmarshalJSON([]byte(out), &params)
-	require.NoError(f.T, err, "out %v\n, err %v", out, err)
+	var params staking.Params
+	require.NoError(f.T, f.Cdc.UnmarshalJSON([]byte(out), &params))
 
 	return params
 }
+
+// QueryStakingDelegation is simcli query staking delegation
+func QueryStakingDelegation(f *cli.Fixtures, from string, valAddr sdk.ValAddress, flags ...string) staking.Delegation {
+	cmd := fmt.Sprintf("%s query staking delegation %s %s %v", f.SimcliBinary, from, valAddr, f.Flags())
+	out, errStr := tests.ExecuteT(f.T, cli.AddFlags(cmd, flags), "")
+	require.Empty(f.T, errStr)
+
+	var delegation staking.Delegation
+	require.NoError(f.T, f.Cdc.UnmarshalJSON([]byte(out), &delegation))
+
+	return delegation
+}
+
+// QueryStakingDelegations is simcli query staking delegations
+func QueryStakingDelegations(f *cli.Fixtures, from string, flags ...string) []staking.Delegation {
+	cmd := fmt.Sprintf("%s query staking delegations %s %v", f.SimcliBinary, from, f.Flags())
+	out, errStr := tests.ExecuteT(f.T, cli.AddFlags(cmd, flags), "")
+	require.Empty(f.T, errStr)
+
+	var delegations []staking.Delegation
+	require.NoError(f.T, f.Cdc.UnmarshalJSON([]byte(out), &delegations))
+
+	return delegations
+}
+
+// QueryStakingRedelegation is simcli query staking redelegation
+func QueryStakingRedelegation(f *cli.Fixtures, delAdrr, srcVal, dstVal string, flags ...string) []staking.RedelegationResponse {
+	cmd := fmt.Sprintf("%s query staking redelegation %v %v %v %v", f.SimcliBinary, delAdrr, srcVal, dstVal, f.Flags())
+	out, errStr := tests.ExecuteT(f.T, cli.AddFlags(cmd, flags), "")
+	require.Empty(f.T, errStr)
+
+	var redelegations []staking.RedelegationResponse
+	require.NoError(f.T, f.Cdc.UnmarshalJSON([]byte(out), &redelegations))
+
+	return redelegations
+}
+
+// QueryStakingRedelegations is simcli query staking redelegation
+func QueryStakingRedelegations(f *cli.Fixtures, delAdrr string, flags ...string) []staking.RedelegationResponse {
+	cmd := fmt.Sprintf("%s query staking redelegations %v %v", f.SimcliBinary, delAdrr, f.Flags())
+	out, errStr := tests.ExecuteT(f.T, cli.AddFlags(cmd, flags), "")
+	require.Empty(f.T, errStr)
+
+	var redelegations []staking.RedelegationResponse
+	require.NoError(f.T, f.Cdc.UnmarshalJSON([]byte(out), &redelegations))
+
+	return redelegations
+}
+
+// QueryStakingRedelegationsFrom is simcli query staking redelegation
+func QueryStakingRedelegationsFrom(f *cli.Fixtures, valAddr string, flags ...string) []staking.RedelegationResponse {
+	cmd := fmt.Sprintf("%s query staking redelegations-from %v %v", f.SimcliBinary, valAddr, f.Flags())
+	out, errStr := tests.ExecuteT(f.T, cli.AddFlags(cmd, flags), "")
+	require.Empty(f.T, errStr)
+
+	var redelegations []staking.RedelegationResponse
+	require.NoError(f.T, f.Cdc.UnmarshalJSON([]byte(out), &redelegations))
+
+	return redelegations
+}
+
+// TODO debug QueryStakingUnbondingDelegation command
```

Comment:
@atheeshp is this still pending?

Reply:
Missed removing the comment, updated it. thanks